superset/migrations/versions/b8d3a24d9131_new_dataset_models.py

```diff
@@ -207,427 +244,557 @@ class NewTable(Base):
+                    ).label("verbose_name"),
+                    TableColumn.python_date_format,
+                    SqlMetric.d3format,
+                    SqlMetric.metric_type,
+                    NewColumn.extra_json,
+                    SqlaTable.is_managed_externally,
+                    SqlaTable.external_url,
+                ]
+            )
+            .select_from(get_joined_tables(offset, limit))
+            .where(
+                # pre-filter to columns with potential updates
+                or_(
+                    NewColumn.is_physical,
+                    TableColumn.groupby.is_(False),
+                    TableColumn.filterable.is_(False),
+                    TableColumn.verbose_name.isnot(None),
+                    TableColumn.verbose_name.isnot(None),
+                    SqlMetric.verbose_name.isnot(None),
+                    SqlMetric.d3format.isnot(None),
+                    SqlMetric.metric_type.isnot(None),
+                )
+            )
+        )
+
+        if SHOW_PROGRESS:
+            start = offset + 1
+            end = min(total, offset + limit)
+            count = session.query(func.count()).select_from(query).scalar()
+            print(f"   Column {start:,} to {end:,}: {count:,} may be updated")
+
+        for (
+            column_id,
+            is_physical,
+            column_name,
+            sqlalchemy_uri,
+            groupby,
+            filterable,
+            verbose_name,
+            python_date_format,
+            d3format,
+            metric_type,
+            extra_json,
+            is_managed_externally,
+            external_url,
+        ) in session.execute(query):
+            try:
+                extra = json.loads(extra_json or "{}")
+            except json.decoder.JSONDecodeError:
+                extra = {}
+            updated_extra = {**extra}
+            updates = {}
+
+            # update expression for physical table columns
+            if is_physical and column_name and sqlalchemy_uri:
+                drivername = sqlalchemy_uri.split("://")[0]
+                if is_physical and drivername:
+                    quoted_expression = get_identifier_quoter(drivername)(column_name)
+                    if quoted_expression != column_name:
+                        updates["expression"] = quoted_expression
+
+            if is_managed_externally:
+                updates["is_managed_externally"] = True
+            if external_url:
+                updates["external_url"] = external_url
+
+            # update extra json
+            if groupby is False:
+                updated_extra["groupby"] = groupby
+            if filterable is False:
+                updated_extra["filterable"] = filterable
+            if verbose_name is not None:
+                updated_extra["verbose_name"] = verbose_name
+            if python_date_format is not None:
+                updated_extra["python_date_format"] = verbose_name
```

Comment:
```suggestion
                updated_extra["python_date_format"] = python_date_format
```
copy paste error?

Reply:
Thanks for the catch! I changed this to a dict to avoid such typo.